### PR TITLE
feat: enable drag and drop feature for import collection

### PR DESCRIPTION
**Pull request by itsmdasifraza**
## Description
enable drag and drop feature for import collection
Fixes issue #112 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

File: src/lib/components/inputs/FileInput.svelte [hunk 1]
```diff
@@ -38,19 +43,39 @@
   ) => {
     onChange(event, maxFileSize, supportedFileTypes);
   };
+
+  const formatSizeUnits = (bytes: number) => {
+    let response: string;
+    if (bytes >= 1073741824) {
```
Comment (LordNayan) [inline src/lib/components/inputs/FileInput.svelte:49]: Try to take this in kb or mb and from a constant.

Comment (LordNayan) [inline src/lib/components/inputs/FileInput.svelte:49]: Can we take this from constant which you may define at the top of the file

Comment (itsmdasifraza) [reply src/lib/components/inputs/FileInput.svelte]: resolved

File: src/lib/components/collections/collections-list/ImportCollection.svelte [hunk 2]
```diff
@@ -31,20 +31,67 @@
...
+    },
+  };
+  const handleLogoInputChange = (
+    e: any,
+    maxSize: number,
+    supportedFileTypes: string[],
+  ) => {
+    if (
+      (e?.target?.files && e?.target?.files[0].size > maxSize * 1024) ||
+      (e?.dataTransfer?.files &&
```
Comment (parthdalaltech) [inline src/lib/components/collections/collections-list/ImportCollection.svelte:57]: if you are using this at multiple places (line 58 also ) extract it to a const and then use it here like const filesize  or extract const filesize function to reuse.

File: src/lib/components/collections/collections-list/ImportCollection.svelte [hunk 3]
```diff
@@ -31,20 +31,67 @@
...
+      uploadCollection.file.showFileTypeError = true;
+      uploadCollection.file.invalid = true;
+      return;
+    }
+    uploadCollection.file.showFileSizeError = false;
+    uploadCollection.file.showFileTypeError = false;
+    uploadCollection.file.invalid = false;
+    uploadCollection.file.value =
+      (e?.target?.files && e?.target?.files[0]) ||
+      (e?.dataTransfer?.files && e?.dataTransfer?.files[0]);
```
Comment (parthdalaltech) [inline src/lib/components/collections/collections-list/ImportCollection.svelte:81]: convert this also to const above to fetch here as direct value.

File: src/lib/components/collections/collections-list/ImportCollection.svelte [hunk 4]
```diff
@@ -283,9 +333,9 @@
   textarea {
     width: 100%;
     resize: vertical;
-    height: 80%;
-    max-height: 100%;
+    height: 100px !important;
```
Comment (parthdalaltech) [inline src/lib/components/collections/collections-list/ImportCollection.svelte:336]: use calc function to set height of the textarea

File: src/lib/components/collections/collections-list/ImportCollection.svelte [hunk 5]
```diff
@@ -283,9 +333,9 @@
   textarea {
     width: 100%;
     resize: vertical;
-    height: 80%;
-    max-height: 100%;
+    height: 100px !important;
     background-color: var(--blackColor);
+    margin-bottom: 15px;
```
Comment (parthdalaltech) [inline src/lib/components/collections/collections-list/ImportCollection.svelte:338]: if possible change to %

File: src/lib/components/collections/collections-list/ImportCollection.svelte [hunk 6]
```diff
@@ -312,30 +362,15 @@
   .container {
     display: flex;
     position: fixed;
-    height: 90%;
-    width: 50%;
     top: 50%;
     left: 50%;
+    max-width: 488px;
```
Comment (parthdalaltech) [inline src/lib/components/collections/collections-list/ImportCollection.svelte:367]: calc function

**Review by parthdalaltech — COMMENTED** [review]
added comments.
also visit this link below for creating a generic function for setting nested object values if using at multiple places.
https://stackoverflow.com/questions/11054922/setting-a-nested-object-value

File: src/lib/components/collections/collections-list/ImportCollection.svelte (hunk 6 above)
Comment (itsmdasifraza) [reply src/lib/components/collections/collections-list/ImportCollection.svelte]: resolved

File: src/lib/components/collections/collections-list/ImportCollection.svelte (hunk 5 above)
Comment (itsmdasifraza) [reply src/lib/components/collections/collections-list/ImportCollection.svelte]: resolved

File: src/lib/components/collections/collections-list/ImportCollection.svelte (hunk 4 above)
Comment (itsmdasifraza) [reply src/lib/components/collections/collections-list/ImportCollection.svelte]: resolved

File: src/lib/components/collections/collections-list/ImportCollection.svelte (hunk 3 above)
Comment (itsmdasifraza) [reply src/lib/components/collections/collections-list/ImportCollection.svelte]: resolved

File: src/lib/components/collections/collections-list/ImportCollection.svelte (hunk 2 above)
Comment (itsmdasifraza) [reply src/lib/components/collections/collections-list/ImportCollection.svelte]: resolved

**Review by parthdalaltech — APPROVED** [review]

**Review by LordNayan — APPROVED** [review]

**Pull request merged**